[2024-10-31T19:19:16.291+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-10-31T19:19:16.386+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T19:19:09.116957+00:00 [queued]>
[2024-10-31T19:19:16.405+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T19:19:09.116957+00:00 [queued]>
[2024-10-31T19:19:16.406+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-10-31T19:19:16.439+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): sensor_data_consumer> on 2024-10-31 19:19:09.116957+00:00
[2024-10-31T19:19:16.475+0000] {standard_task_runner.py:64} INFO - Started process 859 to run task
[2024-10-31T19:19:16.485+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sensor_data_consumer', 'sensor_data_consumer', 'manual__2024-10-31T19:19:09.116957+00:00', '--job-id', '25', '--raw', '--subdir', 'DAGS_FOLDER/***_consumer.py', '--cfg-path', '/tmp/tmpft57nyja']
[2024-10-31T19:19:16.490+0000] {standard_task_runner.py:91} INFO - Job 25: Subtask sensor_data_consumer
[2024-10-31T19:19:16.678+0000] {task_command.py:426} INFO - Running <TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T19:19:09.116957+00:00 [running]> on host c9b652114d41
[2024-10-31T19:19:16.961+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Bich Ly' AIRFLOW_CTX_DAG_ID='sensor_data_consumer' AIRFLOW_CTX_TASK_ID='sensor_data_consumer' AIRFLOW_CTX_EXECUTION_DATE='2024-10-31T19:19:09.116957+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-31T19:19:09.116957+00:00'
[2024-10-31T19:19:16.962+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-10-31T19:19:17.029+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2024-10-31T19:19:17.031+0000] {spark_submit.py:344} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS spark_streaming_job.py
[2024-10-31T19:19:17.406+0000] {spark_submit.py:495} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-10-31T19:19:17.411+0000] {spark_submit.py:495} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/spark-class: line 71: /usr/lib/jvm/java-11-openjdk-amd64/bin/java: No such file or directory
[2024-10-31T19:19:17.417+0000] {spark_submit.py:495} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/spark-class: line 97: CMD: bad array subscript
[2024-10-31T19:19:17.430+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-10-31T19:19:17.466+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 426, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS spark_streaming_job.py. Error code is: 1.
[2024-10-31T19:19:17.486+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=sensor_data_consumer, task_id=sensor_data_consumer, run_id=manual__2024-10-31T19:19:09.116957+00:00, execution_date=20241031T191909, start_date=20241031T191916, end_date=20241031T191917
[2024-10-31T19:19:17.538+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 25 for task sensor_data_consumer (Cannot execute: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS spark_streaming_job.py. Error code is: 1.; 859)
[2024-10-31T19:19:17.582+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2024-10-31T19:19:17.681+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-31T19:19:17.691+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
